From d50205d8d21d72ba26647ccb9e2be0c18a1651be Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Thu, 17 Sep 2015 17:01:37 +0800
Subject: [PATCH] video: rockchip: hdmi: redefine input/output color
 relationship

For YCbCr420 mode, input color and output color must be same as
HDMI_COLOR_YCBCR420.

Change-Id: I302c1a0572d706abdebf5d2d35bd100e0242edf9
Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
(cherry picked from commit c3ed688f02f76dfc5533124c41bbb8c016e3cc97)
---
 drivers/video/rockchip/hdmi/rockchip-hdmi-core.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c b/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
index 3254b39d66af..a7a83265d49c 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
@@ -125,14 +125,14 @@ static void hdmi_wq_set_video(struct hdmi *hdmi)
 		}
 	}
 	pr_info("hdmi output corlor mode is %d\n", video->color_output);
-	video->color_input = HDMI_COLOR_RGB_0_255;
-	if (hdmi->property->feature & SUPPORT_YCBCR_INPUT) {
-		if (video->color_output == HDMI_COLOR_YCBCR444 ||
-		    video->color_output == HDMI_COLOR_YCBCR422)
-			video->color_input = HDMI_COLOR_YCBCR444;
-		else if (video->color_output == HDMI_COLOR_YCBCR420)
-			video->color_input = HDMI_COLOR_YCBCR420;
-	}
+	if ((hdmi->property->feature & SUPPORT_YCBCR_INPUT) &&
+	    (video->color_output == HDMI_COLOR_YCBCR444 ||
+	     video->color_output == HDMI_COLOR_YCBCR422))
+	     video->color_input = HDMI_COLOR_YCBCR444;
+	else if (video->color_output == HDMI_COLOR_YCBCR420)
+		video->color_input = HDMI_COLOR_YCBCR420;
+	else
+		video->color_input = HDMI_COLOR_RGB_0_255;
 
 	if (hdmi->vic & HDMI_VIDEO_DMT) {
 		video->vic = hdmi->vic;
-- 
2.35.3

